GiniVision/Classes/CameraViewController.swift

```diff
@@ -194,14 +209,12 @@ public typealias CameraErrorBlock = (_ error: CameraError) -> ()
     
     public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
         super.viewWillTransition(to: size, with: coordinator)
-        // note that in the mapping left and right are reversed. That's because landscape values have different meanings
-        // in UIDeviceOrientation and AVCaptureVideoOrientation. Refer to their documentaion for more info
-        let orientationsMapping = [UIDeviceOrientation.portrait: AVCaptureVideoOrientation.portrait,
-                                   UIDeviceOrientation.landscapeRight: AVCaptureVideoOrientation.landscapeLeft,
-                                   UIDeviceOrientation.landscapeLeft: AVCaptureVideoOrientation.landscapeRight,
-                                   UIDeviceOrientation.portraitUpsideDown: AVCaptureVideoOrientation.portraitUpsideDown]
+        
         let orientation = UIDevice.current.orientation
         (previewView.layer as? AVCaptureVideoPreviewLayer)?.connection?.videoOrientation = orientationsMapping[orientation]!
+        
+        // Set cameraOverlay image once device has rotated
+        cameraOverlay.image = cameraOverlayImage
```

Comment:
This means that every time you rotate, a new image is generated. It's not ideal.

Also, it's kind of a side effect. You are setting the image, which would normally make no difference, but you are using the side effect of the getter always generating a new image and considering the device orientation.